src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/ListFuture.java

```diff
@@ -37,6 +36,7 @@
    * each component future to fill out the value in the List when that future
    * completes.
    */
+@Deprecated
```

Comment:
This class is only used from workflow-support. Deprecating it so that workflow-support can prefer the upstream version once core is on a recent version of Guava and workflow-support's baseline has been updated.